docs/metrics.md

```diff
@@ -0,0 +1,422 @@
+
+
+### Properties
+
+Details about each property in the payload.
+
+
+#### Survey data
+
+Data gathered from the user in the short survey at the beginning of the report lifecycle.
+
+
+##### `type`
+
+The reason for the submission. String, possible values:
+
+* `random` - if the user was randomly prompted.
+* `user` - if the user initiated the report by the WebExtension page action.
+
+
+##### `sentiment`
+
+The user's feeling about the current page, scored on a five-point Likert scale. Number, between `1` and `5`, inclusive.
+
+
+##### `reason`
+
+The reason the user feels the way they indicated in the `sentiment`. String, possible value _TBD_.
+
+
+##### `details`
+
+Additional details about why the user feels the way they indicated in `sentiment` and `reason`, entered into an open textarea. String or `null`.
+
+  "details": null,
+
+
+#### Browser data
+
+Characteristics of the browser, unrelated to the page or session.
+
+
+##### `telemetryId`
+
+The user's telemetry ID, retrieved from `toolkit.telemetry.cachedClientID`. String, UUID.
+
+
+##### `platform`
+
+The name of the operating system. String, one of the [`OS_TARGET`](https://developer.mozilla.org/docs/Mozilla/Developer_guide/Build_Instructions/OS_TARGET) values.
+
+
+##### `language`
+
+The user's preferred language. String, gotten from [`navigator.language`](https://developer.mozilla.org/docs/Web/API/NavigatorLanguage/language).
+
+
+##### `version`
+
+The version of Firefox, gotten from [`system.version`](https://developer.mozilla.org/Add-ons/SDK/High-Level_APIs/system#version).
+
+
+##### `channel`
+
+The Firefox release channel. String, one of `nightly`, `developer`, `beta`, or `release`.
+
+
+##### `addons`
+
+A list of IDs of installed add-ons. Array.
+
+
+##### `adBlocker`
+
+Does the user have an add-on installed that is known to block ads? Boolean.
+
+
+##### `openTabs`
+
+The count of open tabs across all windows. Number.
+
+
+##### `openWindows`
+
+The number of open windows across the browser. Number.
+
+
+##### `hangs`
+
+The number of recent hangs across the browser. Number.
+
+
+#### Page data
+
+Data about the page from which the report was initiated.
+
+
+##### `timestamp`
+
+The timestamp from which the page request began. Integer.
+
+
+##### `protocol`
+
+The protocol used to retrieve the page, as reported by `window.location.protocol`. String.
+
+
+##### `protocolVersion`
+
+The version of the protocol used to retrieve the page, as reported by the `Version` header. String.
+
+
+##### `trackingProtection`
+
+Is tracking protection enabled for the page? Boolean.
+
+
+##### `timerContentLoaded`
+
+Number of milliseconds between the start of the page load and the `DOMContentLoaded` event. Number.
+
+
+##### `timerWindowLoad`
+
+Number of milliseconds between the start of the page load and the `load` event on `window`. Number.
+
+
+##### `timerFirstPaint`
```

Comment:
`TIME_TO_NON_BLANK_PAINT_MS` has not made it to release yet, we might just be able to measure first paint event.